pkg/controller/direct/dataflow/dataflowflextemplatejob_controller.go

```diff
@@ -330,30 +333,54 @@ func (a *dataflowFlexTemplateJobAdapter) Create(ctx context.Context, createOp *d
 	}
 
 	job := response.GetJob()
-	jobID := job.GetId()
-	// TODO: Use jobCreateTime for mutation checking?
-	// jobCreateTime := job.CreateTime()
 
-	ready := false
-	for !ready {
-		time.Sleep(2 * time.Second)
-		latest, err := a.getJob(ctx, jobID)
-		if err != nil {
-			// TODO: not right!
-			return fmt.Errorf("getting state of job")
+	if err := a.updateStatus(ctx, createOp, job); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (a *dataflowFlexTemplateJobAdapter) updateStatus(ctx context.Context, op directbase.Operation, job *pb.Job) error {
+	status := &krm.DataflowFlexTemplateJobStatus{
+		JobID:        job.GetId(),
+		CurrentState: direct.PtrTo(job.CurrentState.String()),
+		Type:         direct.PtrTo(job.Type.String()),
+	}
+
+	var readyCondition *v1alpha1.Condition
+
+	switch job.CurrentState {
+	case pb.JobState_JOB_STATE_RUNNING:
+		readyCondition = &v1alpha1.Condition{
+			Type:    v1alpha1.ReadyConditionType,
+			Status:  v1.ConditionTrue,
+			Reason:  k8s.UpToDate,
+			Message: "The resource is up to date",
 		}
-		switch latest.CurrentState {
-		case pb.JobState_JOB_STATE_RUNNING:
-			ready = true
-		default:
-			log.Info("unexpected job state waiting for job running", "state", latest.CurrentState)
+
+	case pb.JobState_JOB_STATE_FAILED:
+		readyCondition = &v1alpha1.Condition{
+			Type:    v1alpha1.ReadyConditionType,
+			Status:  v1.ConditionFalse,
+			Reason:  k8s.UpToDate,
+			Message: "Dataflow job is in FAILED state",
+		}
+
+	default:
+		readyCondition = &v1alpha1.Condition{
+			Type:    v1alpha1.ReadyConditionType,
+			Status:  v1.ConditionFalse,
+			Reason:  k8s.Updating,
+			Message: fmt.Sprintf("Waiting for Dataflow job to be running (state is %v)", job.CurrentState),
 		}
 	}
 
-	status := &krm.DataflowFlexTemplateJobStatus{
-		JobID: jobID,
+	if err := op.UpdateStatus(ctx, status, readyCondition); err != nil {
+		return fmt.Errorf("updating status: %w", err)
 	}
-	return setStatus(u, status)
+
+	return nil
 }
 
 // Update implements the Adapter interface.
```

Comment:
Shall we also check the Job status to decide whether to update or not?  
i.e. pb.JobState_JOB_STATE_CANCELLED --> no update and stop reconciling

Reply:
Actually, changed my mind and made it explicit - we now call op.RequestRequeue when we want to requeue.